activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/test/java/org/activiti/cloud/services/query/rest/TaskEntityControllerIT.java

```diff
@@ -250,4 +250,60 @@ public void should_returnTaskPermissions_when_invokeGetTaskById() throws Excepti
             .ofLength(1)
             .thatContains(TaskPermissions.VIEW);
     }
+
+    @Test
+    void testPaginationResolverWithoutSettingMaxOnRequest() throws Exception {
+        //given
+        AlfrescoPageRequest pageRequest = new AlfrescoPageRequest(1001, 1000, PageRequest.of(0, 1000));
+
+        given(taskRepository.findAll(any(), eq(pageRequest)))
+            .willReturn(new PageImpl<>(Collections.singletonList(buildDefaultTask()), pageRequest, 1000));
+
+        //when
+        MvcResult result = mockMvc
+            .perform(get("/v1/tasks").accept(MediaType.APPLICATION_JSON))
+            //then
+            .andExpect(status().isOk())
+            .andReturn();
+
+        assertThatJson(result.getResponse().getContentAsString())
+            .node("list.pagination.skipCount")
+            .isEqualTo(1001)
+            .node("list.pagination.maxItems")
+            .isEqualTo(1000)
+            .node("list.pagination.count")
+            .isEqualTo(1)
+            .node("list.pagination.hasMoreItems")
+            .isEqualTo(false)
+            .node("list.pagination.totalItems")
+            .isEqualTo(1002);
+    }
+
+    @Test
+    void testPaginationResolverWithMaxRequestExceedingLimit() throws Exception {
+        //given
+        AlfrescoPageRequest pageRequest = new AlfrescoPageRequest(1001, 1000, PageRequest.of(0, 1000));
+
+        given(taskRepository.findAll(any(), eq(pageRequest)))
+            .willReturn(new PageImpl<>(Collections.singletonList(buildDefaultTask()), pageRequest, 1000));
+
+        //when
+        MvcResult result = mockMvc
+            .perform(get("/v1/tasks?skipCount=2001&maxItems=2000").accept(MediaType.APPLICATION_JSON))
+            //then
+            .andExpect(status().isOk())
+            .andReturn();
+
+        assertThatJson(result.getResponse().getContentAsString())
+            .node("list.pagination.skipCount")
+            .isEqualTo(1001)
+            .node("list.pagination.maxItems")
+            .isEqualTo(1000)
+            .node("list.pagination.count")
+            .isEqualTo(1)
+            .node("list.pagination.hasMoreItems")
+            .isEqualTo(false)
+            .node("list.pagination.totalItems")
+            .isEqualTo(1002);
+    }
```

Comment:
In any case, we should never touch to `skipCount` that gives how many elements should be left behind before getting the first element to be returned (this is the equivalent of start index). In this specific case, the caller is asking for up to `2000` elements, skipping the first 2001 elements from the results, but what he's getting in return is up `1000` (what could still be fine), skipping only the first 1001 elements from the results: he's getting 1000 elements that he's not expecting.